File: client/go/src/example/connect/sync.go
package connect

import (
	"fmt"
	"context"
	"biostar/service/connect"
)


func (s *ConnectSvc) Connect(deviceIP string, devicePort int, useSSL bool) (uint32, error) {
	req := &connect.ConnectRequest{
		ConnectInfo: &connect.ConnectInfo{
			IPAddr: deviceIP,
			Port: int32(devicePort),
			UseSSL: useSSL,
		},
	}

	resp, err := s.client.Connect(context.Background(), req)

	if err != nil {
		fmt.Printf("Cannot connect to %v:%v: %v\n", deviceIP, devicePort, err)
		return 0, err
	}

	return resp.GetDeviceID(), nil
}


func (s *ConnectSvc) Disconnect(deviceIDs []uint32) error {
	req := &connect.DisconnectRequest{
		DeviceIDs: deviceIDs,
	}

	_, err := s.client.Disconnect(context.Background(), req)

	if err != nil {
		fmt.Printf("Cannot disconnect %v: %v\n", deviceIDs, err)
		return err
	}

	return nil
}


func (s *ConnectSvc) DisconnectAll() error {
	req := &connect.DisconnectAllRequest{
	}

	_, err := s.client.DisconnectAll(context.Background(), req)

	if err != nil {
		fmt.Printf("Cannot disconnect all: %v\n", err)
		return err
	}

	return nil
}

File: client/node/example/connect/sync.js
const connectMessage = require('../../biostar/service/connect_pb');
const connect = require('./connect');

function connectToDevice(addr, port, useSSL) {
  var connInfo = new connectMessage.ConnectInfo();
  connInfo.setIpaddr(addr);
  connInfo.setPort(port);
  connInfo.setUsessl(useSSL);

  var req = new connectMessage.ConnectRequest();
  req.setConnectinfo(connInfo);

  return new Promise((resolve, reject) => {
    connect.getClient().connect(req, (err, response) => {
      if(err) {
        console.error('Cannot connect to the device: ', err)
        reject(err);
        return;
      }

      resolve(response.getDeviceid());
    });
  });
}

function disconnect(deviceIDs) {
  var req = new connectMessage.DisconnectRequest();
  req.setDeviceidsList(deviceIDs);

  return new Promise((resolve, reject) => {
    connect.getClient().disconnect(req, (err, response) => {
      if(err) {
        console.error('Cannot disconnect: ', err)
        reject(err);
        return;
      }

      resolve(response);
    });
  });
}


function disconnectAll() {
  var req = new connectMessage.DisconnectAllRequest();

  return new Promise((resolve, reject) => {
    connect.getClient().disconnectAll(req, (err, response) => {
      if(err) {
        console.error('Cannot disconnect all: ', err)
        reject(err);
        return;
      }

      resolve(response);
    });
  });
}


module.exports.connectToDevice = connectToDevice;
module.exports.disconnect = disconnect;
module.exports.disconnectAll = disconnectAll;